pipeline/pipeline.py

```diff
@@ -0,0 +1,112 @@
+
+
+from pipeline.parameters import Parameters
+from pipeline.data_store import DataStore
+from pipeline.preprocessor import Preprocessor
+from pipeline.astrometry import Astrometry
+from pipeline.calibrator import Calibrator
+from pipeline.subtractor import Subtractor
+from pipeline.detector import Detector
+from pipeline.extractor import Extractor
+
+
+# should this come from db.py instead?
+from models.base import SmartSession
+
+config = {}  # TODO: replace this with Rob's config loader
+
+
+# put all the top-level pipeline parameters in the init of this class:
+class ParsPipeline(Parameters):
+
+        def __init__(self):
+            super().__init__()
+
+            self.add_par('example_pipeline_parameter', 1, int, 'an example pipeline parameter')
+
+            self._enforce_no_new_attrs = True  # lock against new parameters
+
+
+class Pipeline:
+    def __init__(self, **kwargs):
+        # top level parameters
+        self.pars = ParsPipeline(config.get('pipeline', {}))
```

Comment:
I don't think you mean config.get here, but config.value

config.get() method you feed the config filename to in order to get the entire config hierarcy.  I believe here what you're asking for is the "pipeline" subhierarchy of the config object.

Also, config.value doesn't take a default.  We could change that if you want, or we could just mandate that there must be a pipeline (etc.) section in the config.  I'd lean towards the latter, because there are going to be some parameters in some cases that need to be filled in for the pipline to be able to do its work.  WE can always provide a default config with standard and/or dummy values for everything.

Reply:
In fact, it'd be config.Config.get().value('pipeline').  (The "struct" argument to value() should generally not be used; it replaces the entire internal structure with that dictionary.  It's used internally by the config system, but using it from the outside will always be error-prone.)

Reply:
I just assumed `config` is a dictionary with some configuration (I don't think I even merged your config PR into this branch yet). I will have to plug it in with the right interface later.